src/play_controller.cpp

```diff
@@ -528,6 +529,16 @@ void play_controller::do_init_side()
 
 	// Make sure vision is accurate.
 	actions::clear_shroud(current_side(), true);
+	{
+		const auto& active_mods = get_saved_game().classification().active_mods;
+		bool delay_advancements = std::find(active_mods.begin(), active_mods.end(), "delay_advancements") != active_mods.end();
+
+		for(unit &u : resources::gameboard->units()) {
```

Comment:
Why is this loop run unconditionally, when only with delay_advancements active will any units be advances on start-of-turn?

Reply:
because that's 2 lines less i the c++ code for the same effect,  (and the performance implications are negletible)

Reply:
It's 2 lines less at the cost of complicating the conditional, and iterating over all units an extra time at the start of every side turn (regardless of how low that cost may be).

Reply:
well 'at the start of every side turn' is really nothing, keep in mind that every [store_unit] is slower, that we iterate over all units at the end of every user action to check whether the game ended, that we iterate over all units everytimes the player moves the mouse to calculate the resistances of enemies against the units attack in the right-side panel.